src/catalog/src/error.rs

```diff
@@ -218,6 +253,7 @@ impl ErrorExt for Error {
             | Error::FindPartitions { .. }
             | Error::FindRegionRoutes { .. }
             | Error::CacheNotFound { .. }
+            | Error::ViewPlanColumnsChanged { .. }
```

Comment:
Looks like this is closer to `InvalidArg` because the view itself is no longer a valid query.